subsys/bluetooth/host/l2cap.c

```diff
@@ -2653,7 +2653,9 @@ static void l2cap_chan_le_recv(struct bt_l2cap_le_chan *chan,
 		return;
 	}
 
-	l2cap_chan_send_credits(chan, 1);
+	if (bt_l2cap_chan_get_state(&chan->chan) == BT_L2CAP_CONNECTED) {
+		l2cap_chan_send_credits(chan, 1);
+	}
```

Comment:
Suggest to add a comment stating what your commit message says too, otherwise it will be unclear why we need to check this during a `recv` function

Reply:
Just added a comment